corehq/apps/case_migrations/tests/test_restore.py

```diff
@@ -95,14 +93,38 @@ def test_locations_in_restore(self):
         self.assertEqual(response.status_code, 200)
 
         response_content = next(response.streaming_content)
-        locations_content = b""
-        locations = SQLLocation.active_objects.filter(domain=self.domain)
-        for xml_node in FlatLocationSerializer().get_xml_nodes('locations', self.domain, case_id, locations):
-            locations_content += ElementTree.tostring(xml_node, encoding='utf-8')
+
+        locations_content = location_fixture_content.format(
+            user_id=case_id,
+            location_id=location.location_id
+        ).encode('utf-8')
         self.assertIn(locations_content, response_content)
 
     def _generate_restore(self, case_id, user):
         self.client.login(username=user.username, password=user.password)
         url = reverse("migration_restore", args=[self.domain, case_id])
         hmac_header_value = get_hmac_digest(settings.FORMPLAYER_INTERNAL_AUTH_KEY, url)
         return self.client.get(url, HTTP_X_MAC_DIGEST=hmac_header_value)
+
+
+location_fixture_content = (
```

Comment:
that took a while to figure out with newline and empty spaces causing trouble matching xml in response. Open to suggestions if this can be done better.
Note that there are two root notes in this so this is not a valid xml in itself, and is a combination of 2 xml nodes.

Reply:
Ah, you might try this:
https://github.com/dimagi/commcare-hq/blob/1ad0ce5714795c6120906ed1c08f449bfeb617cc/corehq/apps/app_manager/tests/util.py#L23-L28

Check out usages to see how it works.  It wraps the whole thing in `<partial />` so there's only one root node, and I believe it ignores whitespace and such.

Reply:
Thank you, Updated to use that.

The expected XML now needs to have namespace and prefix associated with tags. I guess a side effect to parsing as xml to compare.

Reply:
> The expected XML now needs to have namespace and prefix associated with tags. I guess a side effect to parsing as xml to compare.

That's odd, I hadn't seen that before.  I've mostly used it in the context of suite file generation ([recent example](https://github.com/dimagi/commcare-hq/pull/29036/files#diff-5e51d180dd0fde10d5decd2e10a001042c880b1bdace14e217a7e821a5e11f5bR40)), where you use a path reference to select the node.  I'm not that familiar with the internals though, so I'm not sure why your use-case required namespaces.

Reply:
I think its because the suite xml does not have a namespace/xmlns associated with it whereas a restore does.
Going to merge this now. Good refresher on XML, leaving rest for future.